src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -177,10 +177,11 @@ public Task<RemoteHostClient> TryGetRemoteHostClientAsync(CancellationToken canc
                 var checksums = AddGlobalAssets(cancellationToken);
 
                 // send over global asset
-                await client.RunOnRemoteHostAsync(
-                    WellKnownRemoteHostServices.RemoteHostService, _workspace.CurrentSolution,
-                    nameof(IRemoteHostService.SynchronizeGlobalAssetsAsync),
-                    (object)checksums, cancellationToken).ConfigureAwait(false);
+                await client.IgnoreCancellationAsync(() =>
+                    client.RunOnRemoteHostAsync(
+                        WellKnownRemoteHostServices.RemoteHostService, _workspace.CurrentSolution,
+                        nameof(IRemoteHostService.SynchronizeGlobalAssetsAsync),
+                        (object)checksums, cancellationToken)).ConfigureAwait(false);
```

Comment:
:memo: This one is using a caller-provided cancellation token so it is impacted by my question.

Reply:
answered in different PR - https://github.com/dotnet/roslyn/pull/22968#discussion_r148895378